src/modules/swap/services/safeBundleFlow/safeBundleEthFlow.ts

```
@@ -0,0 +1,140 @@
+        erc20Contract: wrappedNativeContract as unknown as Erc20,
+        spender,
+        amountToApprove: context.inputAmountWithSlippage,
+      })
+
+      txs.push({
+        to: approveTx.to!,
+        data: approveTx.data!,
+        value: '0',
+        operation: 0,
+      })
+    }
+
+    logTradeFlow(LOG_PREFIX, 'STEP 4: post order')
+    const { id: orderId, order } = await signAndPostOrder(orderParams).finally(() => {
+      callbacks.closeModals()
+    })
+
+    addPendingOrderStep(
+      {
+        id: orderId,
+        chainId: context.chainId,
+        order: {
+          ...order,
+          // Initially as hidden, until we get a confirmation from the wallet it succeeded
+          isHidden: true,
+        },
+      },
+      dispatch
+    )
+
+    logTradeFlow(LOG_PREFIX, 'STEP 5: build presign tx')
+    const presignTx = await buildPresignTx({ settlementContract, orderId })
+
+    txs.push({
+      to: presignTx.to!,
+      data: presignTx.data!,
+      value: '0',
+      operation: 0,
+    })
+
+    logTradeFlow(LOG_PREFIX, 'STEP 6: send safe tx')
+
+    const safeTx = await safeAppsSdk.txs.send({ txs })
+
+    logTradeFlow(LOG_PREFIX, 'STEP 7: add safe tx hash and unhide order')
+    partialOrderUpdate(
+      {
+        chainId: context.chainId,
+        order: {
+          id: order.id,
+          // Add Safe tx hash
+          presignGnosisSafeTxHash: safeTx.safeTxHash,
+          // Unhide the order
+          isHidden: false,
+        },
+      },
+      dispatch
+    )
+    tradeFlowAnalytics.sign(swapFlowAnalyticsContext)
+
+    logTradeFlow(LOG_PREFIX, 'STEP 8: add app data to upload queue')
+    callbacks.uploadAppData({ chainId: context.chainId, orderId, appData: appDataInfo })
+
+    logTradeFlow(LOG_PREFIX, 'STEP 9: show UI of the successfully sent transaction')
+    swapConfirmManager.transactionSent(orderId)
+  } catch (error) {
+    logTradeFlow(LOG_PREFIX, 'STEP 10: error', error)
```

Looks like STEP 10 is a required final step (always should throw an error) 😀